.github/workflows/allthekeeps-testnet.yml

```diff
@@ -0,0 +1,103 @@
+name: AllTheKeeps / Testnet
+
+on:
+  push:
+    branches:
+     - allthekeeps-on-ropsten
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment for workflow execution'
+        required: false
+        default: 'dev'
+      upstream_builds:
+        description: 'Upstream builds'
+        required: false
+      upstream_ref:
+        description: 'Git reference to checkout (e.g. branch name)'
+        required: false
+        default: 'main'
+
+jobs:
+  dapp-build-and-publish:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: allthekeeps-on-ropsten
+
+      - name: Load environment variables
+        uses: keep-network/ci/actions/load-env-variables@v1
+        with:
+          environment: 'ropsten'
+      
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+
+      - name: Run install repositories
+        run: ./install-repositories.sh
+      
+      - name: Get upstream packages versions
+        if: github.event_name == 'workflow_dispatch'
+        uses: keep-network/ci/actions/upstream-builds-query@v1
+        id: upstream-builds-query
+        with:
+          upstream-builds: ${{ github.event.inputs.upstream_builds }}
+          query: |
+            keep-core-contracts-version = github.com/keep-network/keep-core/solidity#version
+            keep-ecdsa-contracts-version = github.com/keep-network/keep-ecdsa/solidity#version
+            tbtc-contracts-version = github.com/keep-network/tbtc/solidity#version
+
+      - name: Deploy subgraph to thegraph.com
+        if: github.event_name != 'workflow_dispatch'
```

Comment:
What is a use case for this step? Is it intended to be executed on master merge? Do we expect it to use contract packages versions from `package.json`? I'm not sure about it. Don't we want to publish on `workflow_dispatch` only?

Reply:
In this step we generate a new subgraph for Ropsten purposes and in order to verify that it was generated successfully we deploy it to The Graph. We do not deploy a new version of ATK to GCP in this step, only to The Graph.